### PR TITLE
parser: 'not' functionality on parser

### LINKED ISSUE #24: parser: query parser does not parse all possible combinations of NOT operator correctly
The `NOT` operator is not parsed as NotOp.

OR NOT
-----------
`NOT` is parsed as a `Value`.

**Example:**
`test AND (author:"Alexander, S" OR NOT author:"Costa, M S")`

gives

`AndOp(ValueQuery(Value('test')), AndOp(OrOp(KeywordOp(Keyword('author'), DoubleQuotedValue('Alexander, S')), ValueQuery(Value('NOT'))), KeywordOp(Keyword('author'), DoubleQuotedValue('Costa, M S'))))`

AND (NOT ...
-----------------
`NOT` is parsed as a `Value`.

**Example:**
`test AND (NOT author:"Alexander, S")`

gives

`AndOp(ValueQuery(Value('test')), AndOp(ValueQuery(Value('NOT')), KeywordOp(Keyword('author'), DoubleQuotedValue('Alexander, S'))))`

Only AND NOT works
-----------------------------
NOT is then properly parsed as NotOp

**Example:**

`test AND (author:"Alexander, S" AND NOT author:"Costa, M S")`

gives

`AndOp(ValueQuery(Value('test')), AndOp(KeywordOp(Keyword('author'), DoubleQuotedValue('Alexander, S')), NotOp(KeywordOp(Keyword('author'), DoubleQuotedValue('Costa, M S')))))`

If some operations are not supported like `OR NOT`, then server should return an error. It should not parse it sometime as a `Value` and sometime as `NotOp`.

previously reported as inveniosoftware/invenio#3366

### PR DESCRIPTION
**Pull request by Panos512**
* FIXES Fixes 'not' functionality on the parser.

* Adds test cases with queries using 'not'.
  (closes #24)

Signed-off-by: Panos Paparrigopoulos <panos.paparrigopoulos@cern.ch>

### CODE REVIEW

File: invenio_query_parser/parser.py [hunk 1]
```diff
@@ -25,11 +25,12 @@
...
 from __future__ import absolute_import
 
-import pypeg2
 import re
 
-from pypeg2 import (Keyword, maybe_some, optional, attr,
-                    Literal, omit, some)
+import pypeg2
+
+from pypeg2 import (attr, Keyword, Literal, maybe_some, omit, optional, some)
```
Comment (jirikuncar) [inline invenio_query_parser/parser.py]: please use `isort` to sort imports (`pip install isort; isort invenio_query_parser/parser.py`).

**Comment by jirikuncar** [comment]
@Panos512 please amend `s/FIXES/FIX/`. Can you mention that it fixes parsing of `NOT` operator at the beginning query string and after `OR` operator. Thank :shipit:

**Comment by jirikuncar** [comment]
@Panos512 can you join these 2 bullet points and fix the typo?
```
* FIX Fixes 'not' operator usage at the begining of the query.

* FIX Fixes 'not' operator usage after an 'or' operator.
```
```
* FIX Fixes 'not' operator usage at the beginning of the query
  and after an 'or' operator.
```

**Comment by Panos512** [comment]
@jirikuncar Done! Thank you for the tips :)

**Pull request merged**